File: carelog-be/src/main/java/carelog/carelog/common/config/audit/AuditingDateTimeProvider.java
package carelog.carelog.common.config.audit;

import org.springframework.data.auditing.DateTimeProvider;
import org.springframework.stereotype.Component;

import java.time.OffsetDateTime;
import java.time.temporal.TemporalAccessor;
import java.util.Optional;

@Component("auditingDateTimeProvider")
public class AuditingDateTimeProvider implements DateTimeProvider {

    @Override
    public Optional<TemporalAccessor> getNow() {
        return Optional.of(OffsetDateTime.now());
    }
}

File: src/main/kotlin/carelog/carelog/common/config/audit/AuditingDateTimeProvider.kt
package carelog.carelog.common.config.audit

import org.springframework.data.auditing.DateTimeProvider
import org.springframework.stereotype.Component
import java.time.OffsetDateTime
import java.time.temporal.TemporalAccessor
import java.util.Optional

@Component("auditingDateTimeProvider")
class AuditingDateTimeProvider : DateTimeProvider {

    override fun getNow(): Optional<TemporalAccessor> = Optional.of(OffsetDateTime.now())
}